pallets/asset-registry/src/lib.rs

```diff
@@ -153,6 +147,35 @@ pub mod pallet {
 		}
 	}
 
+	impl<T: Config> Pallet<T> {
+		//Validates that the location points to an asset (Native, Frame based, Erc20) as described
+		// in the xcm-format:  https://github.com/paritytech/xcm-format#concrete-identifiers
+		fn valid_asset_location(location: &MultiLocation) -> bool {
+			let (split_multilocation, last_junction) = location.clone().split_last_interior();
+
+			if let Some(junction) = last_junction {
+				match junction {
```

Comment:
Alex made this refactor using the `maches!` macro which i think makes the code much better in terms of quality:

https://github.com/paritytech/trappist/pull/223/files#diff-d7c7fdfd2cdd4c5fc7b3d6df2550952503740f1819f11331adc24591a8b68e65R81

Reply:
Thanks, this does improve the code and reduces a lot of nesting.